contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,817 @@
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "./../../lib/dappsys/math.sol";
+import "./../colony/ColonyDataTypes.sol";
+import "./../colony/IColony.sol";
+import "./../colonyNetwork/IColonyNetwork.sol";
+import "./../common/ERC20Extended.sol";
+import "./../patriciaTree/PatriciaTreeProofs.sol";
+import "./../tokenLocking/ITokenLocking.sol";
+
+
+contract VotingReputation is DSMath, PatriciaTreeProofs {
+
+  // Events
+  event ExtensionInitialised();
+  event ExtensionDeprecated();
+  event PollCreated(uint256 indexed pollId, uint256 indexed skillId);
+  event PollStaked(uint256 indexed pollId, address indexed staker, uint256 indexed vote, uint256 amount);
+  event PollUnstaked(uint256 indexed pollId, address indexed staker, uint256 indexed vote, uint256 amount);
+  event PollVoteSubmitted(uint256 indexed pollId, address indexed voter);
+  event PollVoteRevealed(uint256 indexed pollId, address indexed voter, uint256 indexed vote);
+  event PollExecuted(uint256 indexed pollId, bytes action, bool success);
+  event PollRewardClaimed(uint256 indexed pollId, address indexed staker, uint256 indexed vote, uint256 amount);
+
+  // Constants
+  uint256 constant UINT256_MAX = 2**256 - 1;
+  uint256 constant UINT128_MAX = 2**128 - 1;
+
+  uint256 constant NAY = 0;
+  uint256 constant YAY = 1;
+
+  uint256 constant CREATE = 0;
+  uint256 constant STAKE_END = 1;
+  uint256 constant SUBMIT_END = 2;
+  uint256 constant REVEAL_END = 3;
+
+  uint256 constant STAKE_COMMIT_FRACTION = (WAD / 4) * 3; // 75%, cannot stake / withdraw after this
+
+  bytes4 constant CHANGE_FUNCTION = bytes4(
+    keccak256("setExpenditureState(uint256,uint256,uint256,uint256,bool[],bytes32[],bytes32)")
+  );
+
+  enum ExtensionState { Deployed, Active, Deprecated }
+
+  // Initialization data
+  ExtensionState state;
+
+  IColony colony;
+  IColonyNetwork colonyNetwork;
+  ITokenLocking tokenLocking;
+  address token;
+
+  uint256 stakeFraction; // Percent of domain reputation needed for staking
+  uint256 minStakeFraction; // Minimum stake as percent of required stake (100% means single-staker)
+
+  uint256 maxVoteFraction; // The percent of total domain rep we need before closing the vote
+  uint256 voterRewardFraction; // Percent of stake paid out to voters as rewards (immediately taken from the stake)
+  uint256 votePowerFraction; // Percent of domain rep used as vote power if no-contest
```

Comment:
What do you mean? They are percentages...

Reply:
> The percent of total domain rep we need before closing the vote

I as a new user read this documentation. I know it needs to be expressed as a WAD. I think "Okay, let's say we need 90% of the vote", and I reasonably conclude that I should put in 90*WAD. i.e. the number 90, expressed as a WAD, because the documentation says I need a percentage.

The correct value I should be putting in is 0.9 denominated as a WAD i.e. 0.9*WAD. 0.9 is not a percentage (when it is expressing nine tenths of the whole), it is a fraction, which numerically differs from the equivalent percentage by a factor of 100.